vowpalwabbit/binary.cc

```diff
@@ -15,14 +16,18 @@
 using namespace VW::config;
 using std::endl;
 
-namespace logger = VW::io::logger;
-
 namespace VW
 {
 namespace binary
 {
+struct binary_data
+{
+  VW::io::logger logger;
+  explicit binary_data(VW::io::logger logger) : logger(std::move(logger)) {}
```

Comment:
```suggestion
  explicit binary_data(VW::io::logger&& logger) : logger(std::move(logger)) {}
```
?

Reply:
is this a copy of the `all.logger`?

Reply:
Yeah copying that object is intended to be the standard usage pattern since it is just a shared pointer